org.ekstep.reviewercomments-1.0/editor/plugin.js

```diff
@@ -0,0 +1,108 @@
+    showComments: function () {
+        var instance = this;
+        var sortedComments = {}
+        if (instance.comments !== undefined && instance.comments.length > 0) {
+            //Display the comments by filtering the stage id from the instance.comments
+            var filtered_comments = _.filter(instance.comments, ['stageId', ecEditor.getCurrentStage().id]);
+            if (filtered_comments.length == 0) {
+                instance.displayNoComments();
+            } else {
+                sortedComments = _.sortBy(filtered_comments, function (dateObj) {
+                    return new Date(dateObj.createdOn);
+                });
+                instance.mapComment(sortedComments);
+            }
+        } else {
+            instance.displayNoComments();
+        }
+    },
+    //To show the 'No review message to the user
+    displayNoComments: function () {
+        jQuery('a[data-content ="comments"]').removeClass('highlight');
+        jQuery('#reviewerComments').html('<div>No review comments</div>');
+    },
+    //Function to format the date 
+    getMonthName: function (date) {
+        var date = new Date(date);
+        return date.toLocaleString('en-us', {
+            month: "long"
+        }).toString().substr(0, 3) + " " + date.getDate();
+    },
+    //To map the comments from the apiResponse
+    mapComment: function (sortedComments) {
+        var instance = this;
+        displayComments = [];
+        var result = _.map(sortedComments, function (item) {
+            var commentThread = {};
+            commentThread.username = item.userInfo.name
+            commentThread.logo = item.userInfo.logo ? item.userInfo.logo : ecEditor.resolvePluginResource(instance.manifest.id, instance.manifest.ver, "assets/reviewer_icon.png");
+            commentThread.username = item.userInfo.name;
+            commentThread.date = instance.getMonthName(item.createdOn);
+            commentThread.body = item.body
+            commentThread.stageId = item.stageId
+            displayComments.push(commentThread);
+        });
+        //update the mapped comments to the template
+        instance.displayStageComments(displayComments);
+    },
+
+    displayStageComments: function (displayComments) {
+        var commentTemplate = _.template(
+            '<% _.each(displayComments, function(item) { %>' +
+            '<div class="comment"> ' +
+            '<div class="avatar"><img src= " <%-item.logo %> "/></div>' +
+            '<div class="content"><div class="flex-class"><span class="author"><%- item.username %></span>' +
+            '<span class="date"><%- item.date %> </span></div>' +
+            '<div class="text"> <%-item.body  %> ' +
+            '</div></div></div>' +
+            '<% }); %>');
+
+        //Add highlight class to the comments tab
+        ecEditor.jQuery('a[data-content ="comments"]').addClass('highlight');
+        ecEditor.jQuery('#reviewerComments').html(commentTemplate);
```

Comment:
@revathitarento  On every `stage:select` event you are updating an DOM Element it's not correct!. Why don't you try with angular ? Please refer `config plugin`.


It's much simpler 
`$scope.comment = " "`
just refer $scope.comments in ur template and inject that in the side bar section and On every change of `stage:select` event w.r.t `stageId` just update the  `$scope.comment` value.

Reply:
@pallakartheekreddy